Installation failed on Windows

**### I am tryin to run 'pip3 install lapjv' on windows but getting this error. Could you help me identify the problem?**







Microsoft Windows [Version 10.0.19042.1466]
(c) Microsoft Corporation. All rights reserved.

C:\Users\User>cd C:\Users\User\Desktop\error-browser-research\lapjv

C:\Users\User\Desktop\error-browser-research\lapjv>pip3 install lapjv
WARNING: Ignoring invalid distribution -ip (c:\python39\lib\site-packages)
WARNING: Ignoring invalid distribution - (c:\python39\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (c:\python39\lib\site-packages)
WARNING: Ignoring invalid distribution - (c:\python39\lib\site-packages)
Collecting lapjv
  Using cached lapjv-1.3.14.tar.gz (10 kB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: numpy>=1.20.0 in c:\users\User\appdata\roaming\python\python39\site-packages (from lapjv) (1.21.4)
Building wheels for collected packages: lapjv
  Building wheel for lapjv (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'C:\Python39\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\User\\AppData\\Local\\Temp\\pip-install-diukxk6m\\lapjv_1f8b6cef705b415d898b52e267b857bb\\setup.py'"'"'; __file__='"'"'C:\\Users\\User\\AppData\\Local\\Temp\\pip-install-diukxk6m\\lapjv_1f8b6cef705b415d898b52e267b857bb\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\User\AppData\Local\Temp\pip-wheel-a_3stb1f'
       cwd: C:\Users\User\AppData\Local\Temp\pip-install-diukxk6m\lapjv_1f8b6cef705b415d898b52e267b857bb\
  Complete output (14 lines):
  running bdist_wheel
  running build
  running build_ext
  building 'lapjv' extension
  creating build
  creating build\temp.win-amd64-3.9
  creating build\temp.win-amd64-3.9\Release
  C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -IC:\Users\User\AppData\Roaming\Python\Python39\site-packages\numpy\core\include -IC:\Python39\include -IC:\Python39\include -IC:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.16.27023\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt /EHsc /Tppython.cc /Fobuild\temp.win-amd64-3.9\Release\python.obj /openmp /std:c++latest /arch:AVX2
  python.cc
  c:\users\User\appdata\local\temp\pip-install-diukxk6m\lapjv_1f8b6cef705b415d898b52e267b857bb\cpuid.h(19): error C3861: '__cpuidex': identifier not found
  c:\users\User\appdata\local\temp\pip-install-diukxk6m\lapjv_1f8b6cef705b415d898b52e267b857bb\cpuid.h(29): error C3861: '__cpuidex': identifier not found
  c:\users\User\appdata\local\temp\pip-install-diukxk6m\lapjv_1f8b6cef705b415d898b52e267b857bb\cpuid.h(33): error C3861: '__cpuidex': identifier not found
  python.cc(127): warning C4244: 'initializing': conversion from 'npy_intp' to 'int', possible loss of data
  error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\BuildTools\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
  ----------------------------------------
  ERROR: Failed building wheel for lapjv
  Running setup.py clean for lapjv
Failed to build lapjv
WARNING: Ignoring invalid distribution -ip (c:\python39\lib\site-packages)
WARNING: Ignoring invalid distribution - (c:\python39\lib\site-packages)
Installing collected packages: lapjv
    Running setup.py install for lapjv ... error
    ERROR: Command errored out with exit status 1:
     command: 'C:\Python39\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\User\\AppData\\Local\\Temp\\pip-install-diukxk6m\\lapjv_1f8b6cef705b415d898b52e267b857bb\\setup.py'"'"'; __file__='"'"'C:\\Users\\User\\AppData\\Local\\Temp\\pip-install-diukxk6m\\lapjv_1f8b6cef705b415d898b52e267b857bb\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\User\AppData\Local\Temp\pip-record-c_nqbwwu\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Python39\Include\lapjv'
         cwd: C:\Users\User\AppData\Local\Temp\pip-install-diukxk6m\lapjv_1f8b6cef705b415d898b52e267b857bb\
    Complete output (14 lines):
    running install
    running build
    running build_ext
    building 'lapjv' extension
    creating build
    creating build\temp.win-amd64-3.9
    creating build\temp.win-amd64-3.9\Release
    C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -IC:\Users\User\AppData\Roaming\Python\Python39\site-packages\numpy\core\include -IC:\Python39\include -IC:\Python39\include -IC:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.16.27023\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt /EHsc /Tppython.cc /Fobuild\temp.win-amd64-3.9\Release\python.obj /openmp /std:c++latest /arch:AVX2
    python.cc
    c:\users\User\appdata\local\temp\pip-install-diukxk6m\lapjv_1f8b6cef705b415d898b52e267b857bb\cpuid.h(19): error C3861: '__cpuidex': identifier not found
    c:\users\User\appdata\local\temp\pip-install-diukxk6m\lapjv_1f8b6cef705b415d898b52e267b857bb\cpuid.h(29): error C3861: '__cpuidex': identifier not found
    c:\users\User\appdata\local\temp\pip-install-diukxk6m\lapjv_1f8b6cef705b415d898b52e267b857bb\cpuid.h(33): error C3861: '__cpuidex': identifier not found
    python.cc(127): warning C4244: 'initializing': conversion from 'npy_intp' to 'int', possible loss of data
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\BuildTools\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\Python39\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\User\\AppData\\Local\\Temp\\pip-install-diukxk6m\\lapjv_1f8b6cef705b415d898b52e267b857bb\\setup.py'"'"'; __file__='"'"'C:\\Users\\User\\AppData\\Local\\Temp\\pip-install-diukxk6m\\lapjv_1f8b6cef705b415d898b52e267b857bb\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\User\AppData\Local\Temp\pip-record-c_nqbwwu\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Python39\Include\lapjv' Check the logs for full command output.
WARNING: Ignoring invalid distribution -ip (c:\python39\lib\site-packages)
WARNING: Ignoring invalid distribution - (c:\python39\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (c:\python39\lib\site-packages)
WARNING: Ignoring invalid distribution - (c:\python39\lib\site-packages)

C:\Users\User\Desktop\error-browser-research\lapjv>

--- a/cpuid.h
+++ b/cpuid.h
@@ -1,6 +1,9 @@
 #pragma once
 
 #ifdef _WIN32
+  #ifdef _MSC_VER
+    #include<intrin.h>
+  #endif
   #define CPUID(info, x)  __cpuidex(info, x, 0)
 #else
   #include <cpuid.h>

